test/compliance/refresh-token-grant-type_test.js

```diff
@@ -0,0 +1,173 @@
+ */
+const OAuth2Server = require('../..');
+const DB = require('../helpers/db');
+const createModel = require('../helpers/model');
+const createRequest = require('../helpers/request');
+const Response = require('../../lib/response');
+
+require('chai').should();
+
+const db = new DB();
+
+const auth = new OAuth2Server({
+  model: createModel(db)
+});
+
+const user = db.saveUser({ id: 1, username: 'test', password: 'test'});
+const client = db.saveClient({ id: 'a', secret: 'b', grants: ['password', 'refresh_token'] });
+const scope = 'read write';
+
+function createLoginRequest () {
+  return createRequest({
+    body: {
+      grant_type: 'password',
+      client_id: client.id,
+      client_secret: client.secret,
+      username: user.username,
+      password: user.password,
+      scope
+    },
+    headers: {
+      'content-type': 'application/x-www-form-urlencoded'
+    },
+    method: 'POST',
+  });
+}
+
+function createRefreshRequest (refresh_token) {
+  return createRequest({
+    method: 'POST',
+    body: {
+      grant_type: 'refresh_token',
+      client_id: client.id,
+      client_secret: client.secret,
+      refresh_token,
+      scope
+    },
+    headers: {
+      'content-type': 'application/x-www-form-urlencoded'
+    }
+  });
+}
+
+describe('RefreshTokenGrantType Compliance', function () {
+  describe('With scope', function () {
+    it('Should generate token response', async function () {
+      const request = createLoginRequest();
+      const response = new Response({});
+
+      const credentials = await auth.token(request, response, {});
+
+      const refreshRequest = createRefreshRequest(credentials.refreshToken);
+      const refreshResponse = new Response({});
+
+      const token = await auth.token(refreshRequest, refreshResponse, {});
+
+      refreshResponse.body.token_type.should.equal('Bearer');
+      refreshResponse.body.access_token.should.equal(token.accessToken);
+      refreshResponse.body.refresh_token.should.equal(token.refreshToken);
+      refreshResponse.body.expires_in.should.be.a('number');
+      refreshResponse.body.scope.should.equal(scope);
+
+      token.accessToken.should.be.a('string');
+      token.refreshToken.should.be.a('string');
+      token.accessTokenExpiresAt.should.be.a('date');
+      token.refreshTokenExpiresAt.should.be.a('date');
+      token.scope.should.equal(scope);
+
+      db.accessTokens.has(token.accessToken).should.equal(true);
+      db.refreshTokens.has(token.refreshToken).should.equal(true);
+    });
+
+    it('Should throw invalid_grant error', async function () {
+      const request = createRefreshRequest('invalid');
+      const response = new Response({});
+
+      await auth.token(request, response, {})
+        .then(() => {
+          throw Error('Should not reach this');
+        }).catch(err => {
+          err.name.should.equal('invalid_grant');
+        });
+    });
+
+    // TODO: test refresh token with different scopes
```

Comment:
This is depending on #104 right?

Reply:
Yes, at the moment it's not throwing anything at all, so the test failed

Reply:
Can you please add #104 to the TODO comment so it's clear where it belongs, just in case

Reply:
Please don't forget the comment to add `related: #104` to this

Reply:
Comment was already added in a previous commit